src/views/virtualmachines/details/VirtualMachineNavPage.tsx

```diff
@@ -8,62 +8,72 @@ import {
 import HorizontalNavbar from '@kubevirt-utils/components/HorizontalNavbar/HorizontalNavbar';
 import { SidebarEditorProvider } from '@kubevirt-utils/components/SidebarEditor/SidebarEditorContext';
 import { getResourceDetailsTitle } from '@kubevirt-utils/constants/page-constants';
+import useK8sWatchData from '@kubevirt-utils/hooks/useK8sWatchData';
+import useMemoizedParams from '@kubevirt-utils/hooks/useMemoizedParams';
+import { VirtualMachineModelGroupVersionKind } from '@kubevirt-utils/models';
 import { getName } from '@kubevirt-utils/resources/shared';
 import useInstanceTypeExpandSpec from '@kubevirt-utils/resources/vm/hooks/useInstanceTypeExpandSpec';
 import { isEmpty } from '@kubevirt-utils/utils/utils';
-import { DocumentTitle, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { DocumentTitle } from '@openshift-console/dynamic-plugin-sdk';
 
 import { useVirtualMachineTabs } from './hooks/useVirtualMachineTabs';
 import VirtualMachineNavPageTitle from './VirtualMachineNavPageTitle';
 
 import './virtual-machine-page.scss';
 
-export type VirtualMachineDetailsPageProps = {
-  kind: string;
-  name: string;
-  namespace: string;
-};
+const VirtualMachineNavPage: FC = () => {
+  const {
+    cluster,
+    name,
+    ns: namespace,
+  } = useMemoizedParams<{
+    cluster?: string;
+    name: string;
+    ns: string;
+  }>();
 
-const VirtualMachineNavPage: React.FC<VirtualMachineDetailsPageProps> = ({
-  kind,
-  name,
-  namespace,
-}) => {
-  const [vm, isLoaded, loadError] = useK8sWatchResource<V1VirtualMachine>(
+  const [vm, isLoaded, loadError] = useK8sWatchData<V1VirtualMachine>(
     runningTourSignal.value
       ? null
       : {
-          kind,
+          cluster,
+          groupVersionKind: VirtualMachineModelGroupVersionKind,
           name,
           namespace,
         },
   );
 
-  const vmToShow = runningTourSignal.value ? tourGuideVM : vm;
+  const vmToShow = useMemo(() => (runningTourSignal.value ? tourGuideVM : vm), [vm]);
 
   const [instanceTypeExpandedSpec, expandedSpecLoading] = useInstanceTypeExpandSpec(vmToShow);
 
   const pages = useVirtualMachineTabs();
 
-  return (
-    <SidebarEditorProvider>
-      <DocumentTitle>{getResourceDetailsTitle(getName(vmToShow), 'VirtualMachine')}</DocumentTitle>
-      <VirtualMachineNavPageTitle
-        instanceTypeExpandedSpec={instanceTypeExpandedSpec}
-        isLoaded={isLoaded || !isEmpty(loadError)}
-        name={name}
-        vm={vmToShow}
-      />
-      <div className="VirtualMachineNavPage--tabs__main">
-        <HorizontalNavbar
-          error={loadError}
+  return useMemo(
```

Comment:
it seems we wrapped entire component in `useMemo` - it would be cleaner to use the recommended  method of memoization - `memo` function.

Reply:
i should create a new component and wrap it with `memo` so its the same

Reply:
why do we need useMemo here?

Reply:
because `useParams` triggers re-rendering